src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp

```diff
@@ -97,20 +98,27 @@ void noteManager::RunCommonTasks()
 	Cyclic();
 	ResetLauncherAngle();
 	ResetElevator();
-
-	Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Sensors"), string("Front"), getfrontIntakeSensor()->Get());
-	Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Sensors"), string("Back"), getbackIntakeSensor()->Get());
-	Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Sensors"), string("Feeder"), getfeederSensor()->Get());
-	Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Sensors"), string("Launcher"), getlauncherSensor()->Get());
+	SetManualLaunchTarget();
+
+	bool protectLauncher = ((m_noteManager->GetCurrentState() != m_noteManager->STATE_READY_AUTO_LAUNCH) ||
```

Comment:
the logic does not seem correct. Let's say the state is LOW_PASS.... the 1st part of the expr (currentState != READY_AUTO_LAUNCH) will be true, setting protectLauncher to true

I would write:
protectLauncher = !( OR all the states in which you do not want to go to safe position)

Reply:
You are right, but we initially had it like that. However the compiler was throwing warning beucase you can't do an or operation on enums. We will take a look tonight and see if we can do it another way

Reply:
Oh I am sorry I think I see what you are saying, basically write !(m_noteManager->GetCurrentState() == Pass || m_noteManager->GetCurrentState() == lowPass ect)